### PR TITLE
docs(main): update router doc

### PR DESCRIPTION
**Pull request by caohuilin**
### <samp>🤖 Generated by Copilot at 5ef558e</samp>

This pull request improves the documentation of the routing features of the `modern.js` framework, both in English and Chinese. It adds new sections, examples, notes, and links to explain the routing modes, conventions, and options, as well as the error boundary support for routes. It also fixes some typos, grammar, and style issues in the existing documentation. The changes affect the files `routes.mdx` and `router.mdx` in the `packages/document/main-doc/docs` directory.

### <samp>🤖 Generated by Copilot at 5ef558e</samp>

*  Add error boundary support for routes by documenting the `useRouteError` hook and the `error.tsx` file ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-e1d71a64ebc45de163a861e84900312c13bbd430ec1d3a69c595cb80fde078dfR110-R131), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L386-R487), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L406-R512), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-785138781be360dd27dd3ba6334f3b5d12866d45c32d4d154fbc2f06ff787f43R109-R130), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L338-R338), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L344-R344))
* Add support for dynamic optional routing, catch-all routing, no-path layout, and no layout by documenting the file naming conventions and the UI layout effects ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L134-R199), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L156-R207), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L182-R243), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L210-R263), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L176-R176), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L202-R202), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L225-R225), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L248-R248))
* Add support for loading by documenting the `loading.tsx` file and the `Suspense` component ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L246-R344), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L269-R386), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L281-R398), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L309-R422), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L309-R310))
* Add support for redirect by documenting the `data loader` file and the `redirect` object ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L332-R475), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L320-R320))
* Add support for runtime configuration by documenting the `config` function and the `AppConfig` object ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L425-R522), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L362-R362))
* Improve the documentation style and consistency by adding or removing code block annotations, quotation marks, links, and capitalization ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L8-R20), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L21-R21), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L157-R157), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L284-R284), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L292-R292), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L300-R300), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L450-R454), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L460-R466), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L472-R472), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L487-R487))
* Improve the documentation clarity and accuracy by rewriting or rephrasing the explanations of the routing file convention, the UI layout, and the `Outlet` component ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L32-R58), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L48-R66), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L68-R105), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L89-R119), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L97-R127), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L105-R135), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L114-R154))
* Improve the documentation completeness and readability by adding example code snippets for the `routes/layout.tsx`, `$.tsx`, `error.tsx`, and `loading.tsx` files ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L56-R74), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L165-R220), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L200-R251), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L226-R321), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L406-R512), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L269-R386), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L281-R398))
* Improve the documentation grammar and correctness by fixing typos and adding missing articles ([link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L56-R74), [link](https://github.com/web-infra-dev/modern.js/pull/3722/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L105-R135))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 1]
```diff
@@ -243,15 +333,13 @@ export default () => {
...
-In each layer directory under `routes/`, the developer can also define a `error.tsx` file, and export a `<ErrorBoundary>` component by default.
-
-When the component exists in a routing directory, the component render error is caught by the `ErrorBoundary` component. The `<ErrorBoundary>` component does not take effect when the directory does not have a `layout.tsx` file defined.
+In each directory under `routes/`, developers can also define an `error.tsx` file that exports an `<ErrorBoundary>` component by default.
 
-`<ErrorBoundary>` can return the UI view when the error occurred. When the `<ErrorBoundary>` component is not declared at the current level, the error will bubble up to the higher component until it is caught or throws an error. At the same time, when a component fails, it will only affect the routed component and sub-component that caught the error. The state and view of other components are not affected and can continue to interact.
+When this component exists in a route directory, any rendering errors will be caught by the `ErrorBoundary` component. When the `layout.tsx` file is not defined in the directory, the `<ErrorBoundary>` component will not take effect.
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:338]: Maybe "in the routes directory" should be used here?

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 2]
```diff
@@ -131,29 +160,51 @@ With a file directory named `[]`, the generated route will be used as a dynamic
...
+    │   └── [id$]
+    │       └── page.tsx
+    ├── blog
+    │   └── page.tsx
+    └── page.tsx
+```
+
+The `routes/user/[id$]/page.tsx` file will be converted to the `/user/:id?` route. All routes under `/user` will match this route, and the `id` parameter is optional. This route is usually used to distinguish between **creation** and **editing**.
+
+In the component, you can use [useParams](/apis/app/runtime/router/router#useparams) to obtain the corresponding named parameter.
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:185]: maybe we can use 'get' instead 'obtain'

Comment (zllkjc) [reply packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:185]: "Obtain" usually mead get + effort ?

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 3]
```diff
@@ -131,29 +160,51 @@ With a file directory named `[]`, the generated route will be used as a dynamic
...
+In the loader, params will be passed as the input parameter of the [loader function](/guides/basic-features/data-fetch#loader-function), and you can get the parameter value through `params.xxx`.
 
-### Catch all routing
+### Catch-all Routing
 
-If a `$.tsx` file is created in the routes directory, this file will act as a wildcard route component that will be rendered when there is no matching route.
+If you create a `$.tsx` file under the routes directory, it will be treated as the catch-all routing component. When there is no matching route, this component will be rendered.
 
 :::note
-`$.tsx` can be thought of as a special `page` routing component that renders `$.tsx` as a child of `layout` when there is a `layout` component in the current directory.
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:143]: use [`<Layout>` component] or [`layout.tsx` file]

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 4]
```diff
@@ -162,11 +213,11 @@ const params = useParams();
 params['*']; // => 'aaa/bbb'
 ```
 
-### Layout with No Path
+### No-path Layout
 
-When a directory name begins with `__`, the corresponding directory name is not converted to the actual routing path, such as the following file directory:
+When the directory name starts with `__`, the corresponding directory name will not be converted to an actual route path. For example, the following file directory:
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:218]: "corresponding" seems redundant here.

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 5]
```diff
@@ -329,51 +441,37 @@ export const init = (context: RuntimeContext) => {
...
+However, this also brings a problem: if the chunk corresponding to the route has not finished loading when the user visits the route, a white screen will appear.
 
-In this case you can show a custom `loading` component by defining a `loading` component before the static resource is loaded.
+In this case, you can define a `Loading` component to display a custom `Loading` component before the static resources are loaded.
 
-To further improve the user experience and reduce time of loading, Modern.js supports defining the `prefetch` property on the Link component to load static resources and data in advance, with three optional values for the `prefetch` property:
+To further improve the user experience and reduce loading time, Modern.js supports defining the `prefetch` attribute on the Link component to preload static resources and data. The `prefetch` attribute has three optional values:
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:452]: "The `prefetch` attribute has three optional values" should be moved down.

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 6]
```diff
@@ -403,16 +501,14 @@ export default () => {
...
 
 :::note
-Modern.js has a series of resource loading and rendering optimizations to the default convention-based routing, and provides out-of-the-box SSR capabilities, when using self-directed routing, need to be packaged by the developer, and it is recommended that developers use convention-based routing.
+Modern.js provides a series of optimizations for resource loading and rendering for conventional routing by default, and provides out-of-the-box SSR capabilities. When using self-controlled routing, developers need to encapsulate these capabilities themselves. It is recommended to use conventional routing.
 :::
 
-use self-controller routing, if the developer turns off the [`runtime.router`](/configure/app/runtime/router) configuration and uses `react-router-dom` directly, then you need to wrap the `Provider` according to the React Router documentation.
-
-```tsx title="src/App.tsx"
+When using self-controlled routing, if the developer turns off the [`runtime.router`](/configure/app/runtime/router) configuration and uses `react-router-dom` directly, they also need to wrap it according to the React Router documentation.
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:507]: This explanation should be placed in the "Other" section, and the "Other" section should describe the differences between `router: true` and `router: false`. (Modern.js will init react-router and reexport apis when config `router: true`)

**Pull request merged**